contracts/test/strategies/curve-amo-ousd.mainnet.fork-test.js

```diff
@@ -0,0 +1,1199 @@
+          attackerUsdtBalanceAfter.sub(attackerUsdtBalanceBefore),
+          6
+        )} USDT and ${formatUnits(
+          attackerOusdBalanceAfter.sub(attackerOusdBalanceBefore)
+        )} OUSD`
+      );
+
+      // Rebase to lock in the profits
+      await ousdVault.rebase();
+      const dataAfterRebase = await snapData();
+      logSnapData(dataAfterRebase, "\nAfter rebase to lock in profits");
+      await logProfit(dataBeforeAttack);
+
+      // Remove all funds from the Curve AMO strategy
+      await ousdVault
+        .connect(impersonatedAMOGovernor)
+        .withdrawAllFromStrategy(curveAMOStrategy.address);
+      const dataAfterStratWithdrawAll = await snapData();
+      logSnapData(
+        dataAfterStratWithdrawAll,
+        "\nAfter withdraw all from strategy"
+      );
+      const finalProfit = await logProfit(dataAfterRebase);
+      expect(finalProfit).to.be.gte(0);
+    });
+
+    it("Should withdraw from strategy", async () => {
+      await balancePool();
+      await mintAndDepositToStrategy();
+
+      const impersonatedVaultSigner = await impersonateAndFund(
+        ousdVault.address
+      );
+
+      const checkBalanceBefore = await curveAMOStrategy.checkBalance(
+        usdt.address
+      );
+      const gaugeBalanceBefore = await curveGauge.balanceOf(
+        curveAMOStrategy.address
+      );
+
+      const amountToWithdraw = defaultDeposit.mul(999).div(1000); // 1000 OUSD
+      await curveAMOStrategy
+        .connect(impersonatedVaultSigner)
+        .withdraw(ousdVault.address, usdt.address, amountToWithdraw.div(1e12));
+
+      expect(
+        checkBalanceBefore.sub(
+          await curveAMOStrategy.checkBalance(usdt.address)
+        )
+      ).to.approxEqualTolerance(amountToWithdraw.mul(2).div(1e12));
+      expect(
+        gaugeBalanceBefore.sub(
+          await curveGauge.balanceOf(curveAMOStrategy.address)
+        )
+      ).to.approxEqualTolerance(amountToWithdraw.mul(2));
+      expect(await ousd.balanceOf(curveAMOStrategy.address)).to.equal(0);
+      expect(await usdt.balanceOf(curveAMOStrategy.address)).to.equal(
+        ousdUnits("0")
+      );
+    });
+
+    it("Should withdraw all from strategy", async () => {
+      await balancePool();
+      await mintAndDepositToStrategy();
+
+      const balanceVault = await usdt.balanceOf(ousdVault.address);
+
+      await curveAMOStrategy.connect(impersonatedVaultSigner).withdrawAll();
+
+      expect(
+        await curveAMOStrategy.checkBalance(usdt.address)
+      ).to.approxEqualTolerance(0);
+      expect(
+        await curveGauge.balanceOf(curveAMOStrategy.address)
+      ).to.approxEqualTolerance(0);
+      expect(await ousd.balanceOf(curveAMOStrategy.address)).to.equal(0);
+      expect(await usdt.balanceOf(curveAMOStrategy.address)).to.equal(
+        ousdUnits("0")
+      );
+      expect(await usdt.balanceOf(ousdVault.address)).to.approxEqualTolerance(
+        balanceVault.add(defaultDeposit.div(1e12))
+      );
+    });
```

Comment:
Can a second `withdrawAll` be added so we can test withdrawal from an empty strategy will not revert
```js
      // Should be able to withdraw all from an empty strategy
      await curveAMOStrategy.connect(impersonatedVaultSigner).withdrawAll();
```

Reply:
Added in this commit: [2ebf607](https://github.com/OriginProtocol/origin-dollar/pull/2436/commits/2ebf607d9fc866f8b46d0a329a6ef67de5ca0546)